src/audio/kpb.c

```diff
@@ -1417,6 +1434,47 @@ static int kpb_buffer_data(struct comp_dev *dev,
 	return ret;
 }
 
+/**
+ * \brief Unregister clients in the system.
+ *
+ * \param[in] kpb - kpb device component pointer.
+ * \param[in] cli - pointer to KPB client's data.
+ *
+ * \return integer representing either:
+ *	0 - success
+ *	-EINVAL - failure.
+ */
+static int kpb_unregister_client(struct comp_data *kpb, struct kpb_client *cli)
+{
+	int ret = 0;
+
+	comp_cl_info(&comp_kpb, "kpb_unregister_client()");
+
+	if (!cli) {
+		comp_cl_err(&comp_kpb, "kpb_unregister_client(): no client data");
+		return -EINVAL;
+	}
+	if (kpb->kpb_no_of_clients >= KPB_MAX_NO_OF_CLIENTS ||
```

Comment:
this shouldn't happen, right? Such a kpb couldn't have been created? But I see that the same is checked in `kpb_register_client(()`

Reply:
Such a kpb should not be created